apps/marketplace/components/BuyerRFX/BuyerRFXEvaluationCriteriaStage.js

```diff
@@ -0,0 +1,222 @@
+/* eslint-disable react/no-array-index-key */
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+import { actions, Form } from 'react-redux-form'
+import formProps from 'shared/form/formPropsSelector'
+import Textarea from 'shared/form/Textarea'
+import Textfield from 'shared/form/Textfield'
+import AUheadings from '@gov.au/headings/lib/js/react.js'
+import CheckboxDetailsField from 'shared/form/CheckboxDetailsField'
+import ErrorAlert from './ErrorAlert'
+import styles from './BuyerRFXEvaluationCriteriaStage.scss'
+
+export const noEmptyWeightings = formValues =>
+  !formValues.includeWeightings || formValues.evaluationCriteria.every(val => val.weighting)
+
+export const weightingsAddUpTo100 = formValues =>
+  !formValues.includeWeightings ||
+  !noEmptyWeightings(formValues) ||
+  formValues.evaluationCriteria.reduce(
+    (accumulator, currentValue) => accumulator + parseInt(currentValue.weighting, 10),
+    0
+  ) === 100
+
+export const noZeroWeightings = formValues =>
+  !formValues.includeWeightings || formValues.evaluationCriteria.every(val => parseInt(val.weighting, 10) > 0)
+
+export const noEmptyCriteria = formValues => formValues.evaluationCriteria.every(val => val.criteria)
+
+class BuyerRFXEvaluationCriteriaStage extends Component {
+  constructor(props) {
+    super(props)
+    this.handleIncludeWeightingsChange = this.handleIncludeWeightingsChange.bind(this)
+    this.handleAddCriteriaClick = this.handleAddCriteriaClick.bind(this)
+    this.removeCriteria = this.removeCriteria.bind(this)
+  }
+
+  getRemainingWeighting() {
```

Comment:
This needs some extra checks on the input.  I stumbled across a negative number.
<img width="823" alt="screen shot 2018-12-15 at 12 26 05 pm" src="https://user-images.githubusercontent.com/2645932/50037442-62280e80-0065-11e9-9878-12af7bb389bb.png">

Reply:
This is by design.